src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -19,16 +19,42 @@ const ApplicationLauncher = _.flow(
   Utils.forwardRefWithName('ApplicationLauncher'),
   wrapWorkspace({
     breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
-    title: _.get('application')
+    title: _.get('application'),
+    activeTab: appLauncherTabName
   })
-)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
+)(({ name, refreshRuntimes, runtimes, persistentDisks, application, workspace, workspace: { workspace: { googleProject, bucketName } } }, ref) => {
   const cookieReady = Utils.useStore(cookieReadyStore)
   const [showCreate, setShowCreate] = useState(false)
   const [busy, setBusy] = useState(false)
+  const [isWelderInit, setIsWelderInit] = useState(application !== 'RStudio') // We've already init Welder if app is Jupyter.
 
   const runtime = getCurrentRuntime(runtimes)
   const runtimeStatus = getConvertedRuntimeStatus(runtime) // preserve null vs undefined
 
+  useEffect(() => {
+    const runtime = getCurrentRuntime(runtimes)
+    const runtimeStatus = getConvertedRuntimeStatus(runtime)
+
+    const setupWelder = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('Error setting up analysis')
```

Comment:
In this case, the error message would specifically correspond to file syncing. Therefore, it may be better to reflect that in the message with something like

```suggestion
      withErrorReporting('Error setting up analysis file syncing')
```